### DISCUSSION TITLE
Android devices with random MAC addresses show multiple times?

### DISCUSSION THREAD

**Discussion by jmigual** [netalertx/NetAlertX#368] [Q&A]
So, in my network I have a different number of Android devices connected to WiFi. These devices change MAC address from time to time and get assigned different IPs. However, I get new copies of these devices every time they change MAC addresses. Is there a way to fix this?

**Comment by jokob-sk** [comment]
Hi there!

You can disable random mac addresses on your network on your physical device. Here is an older guide:

https://github.com/jokob-sk/Pi.Alert/blob/main/docs/RANDOM_MAC.md

Hope this helps,
J

**Reply by jmigual** [reply]
So, is the only solution to disable random MAC addresses on the devices? Can't Pi.Alert track them?

**Reply by jokob-sk** [reply]
Right now yes. The next version of PiAlert will separate out arp-scan into a plugin that should theoretically give more flexibility in tracking and matching decisions (intentionally based on IP addresses). Currently, with my workload, life, and other things this is a very low priority. Also, most of PiAlert depends on MAC addresses being the ID of devices, which makes sense. 

You could look into using other plugins to track devices, such as PiHole, but I haven't tested it.

That being said a donation might convince me to increase the priority of this. Alternatively, you can look into contributing via a PR to this project as well. 


Related issues are: https://github.com/jokob-sk/Pi.Alert/discussions/264

There is one more related issue but I can't find it now and am in a rush.

**Reply by jokob-sk** [reply]
Thinking more about this you could write a plugin that would merge potentially duplicate devices that is executed after a scan... The difficulty is to determine matching criteria. Look into the Plugin system if you want to give it a shot 👍
https://github.com/jokob-sk/Pi.Alert/tree/main/front/plugins

**Reply by jmigual** [reply]
Thanks! I may take a look. For me, I think the name would be the only possible matching criteria but this may not work for everyone:
![image](https://github.com/jokob-sk/Pi.Alert/assets/6493651/029cca4e-91b9-4d11-8cd5-792605947f82)
